dotnet/samples/Demos/VectorStoreRAG/Program.cs

```diff
@@ -0,0 +1,121 @@
+            appConfig.RagConfig.CollectionName,
+            appConfig.AzureCosmosDBMongoDBConfig.ConnectionString,
+            appConfig.AzureCosmosDBMongoDBConfig.DatabaseName);
+        break;
+    case "AzureCosmosDBNoSQL":
+        kernelBuilder.AddAzureCosmosDBNoSQLVectorStoreRecordCollection<TextSnippet<string>>(
+            appConfig.RagConfig.CollectionName,
+            appConfig.AzureCosmosDBNoSQLConfig.ConnectionString,
+            appConfig.AzureCosmosDBNoSQLConfig.DatabaseName);
+        break;
+    case "Qdrant":
+        kernelBuilder.AddQdrantVectorStoreRecordCollection<Guid, TextSnippet<Guid>>(
+            appConfig.RagConfig.CollectionName,
+            appConfig.QdrantConfig.Host,
+            appConfig.QdrantConfig.Port,
+            appConfig.QdrantConfig.Https,
+            appConfig.QdrantConfig.ApiKey);
+        break;
+    case "Redis":
+        kernelBuilder.AddRedisJsonVectorStoreRecordCollection<TextSnippet<string>>(
+            appConfig.RagConfig.CollectionName,
+            appConfig.RedisConfig.ConnectionConfiguration);
+        break;
+    case "Weaviate":
+        kernelBuilder.AddWeaviateVectorStoreRecordCollection<TextSnippet<Guid>>(
+            // Weaviate collection names must start with an upper case letter.
+            char.ToUpper(appConfig.RagConfig.CollectionName[0], CultureInfo.InvariantCulture) + appConfig.RagConfig.CollectionName.Substring(1),
+            null,
+            new() { Endpoint = new Uri(appConfig.WeaviateConfig.Endpoint) });
+        break;
+    default:
+        throw new NotSupportedException($"Vector store type '{appConfig.RagConfig.VectorStoreType}' is not supported.");
+}
+
+// Register all the other required services.
+switch (appConfig.RagConfig.VectorStoreType)
+{
+    case "AzureAISearch":
+    case "AzureCosmosDBMongoDB":
+    case "AzureCosmosDBNoSQL":
+    case "Redis":
+        RegisterServices<string>(builder, kernelBuilder, appConfig);
+        break;
+    case "Qdrant":
+    case "Weaviate":
+        RegisterServices<Guid>(builder, kernelBuilder, appConfig);
+        break;
+    default:
+        throw new NotSupportedException($"Vector store type '{appConfig.RagConfig.VectorStoreType}' is not supported.");
+}
+
+// Build and run the host.
+using IHost host = builder.Build();
+await host.RunAsync().ConfigureAwait(false);
+
+static void RegisterServices<TKey>(HostApplicationBuilder builder, IKernelBuilder kernelBuilder, ApplicationConfig vectorStoreRagConfig)
+    where TKey : notnull
+{
+    // Add a text search implementation that uses the registered vector store record collection for search.
+    kernelBuilder.AddVectorStoreTextSearch<TextSnippet<TKey>>(
+        new TextSearchStringMapper((result) => (result as TextSnippet<TKey>)!.Text!),
+        new TextSearchResultMapper((result) =>
+        {
+            var castResult = result as TextSnippet<TKey>;
+            return new TextSearchResult(value: castResult!.Text!) { Link = castResult.ReferenceLink };
```

Comment:
We are using `Name` in the handlebars template but we don't have one

Reply:
Thanks, removed.